packages/origin-api/src/app.module.ts

```diff
@@ -1,14 +1,35 @@
 import {
   AppModule as ExchangeModule,
   entities as ExchangeEntities,
+  IExchangeConfigurationService,
+  IOrderMapperService,
 } from "@energyweb/exchange";
```

Comment:
Does it help if you use this instead?
```js
import {
  AppModule as ExchangeModule,
  entities as ExchangeEntities,
} from "@energyweb/exchange-irec";
```

Then you might not need to specify the providers at the end of the file.

Reply:
I believe entities should be imported from general `exchange` right? ;)

Anyway, no, unfortunately now I'm missing `IExternalDeviceService` implementation, which is required by irec `ProductService` in irec.

Neither irec or base exchange implement that interface. In demo app it's implemented manually.

Reply:
And this manual implementation is based on `@energyweb/origin-device-registry-irec-form-api` DeviceService.